### DISCUSSION TITLE
Why StencilState is not Copy

### DISCUSSION THREAD

**Discussion by chubei-oppen** [gfx-rs/wgpu#2099] [Q&A] [answered]
Looking at the [source](https://github.com/gfx-rs/wgpu/blob/a59a627f124d03b34c826de64b77f5bf74d6af43/wgpu-types/src/lib.rs#L2041), all its fields are `Copy`. Is there a special reason `StencilState` does not derive `Copy`?

**Comment by kvark** [comment, marked as answer]
IIRC, we picked a threshold - say 16 bytes - and only put `Copy` where the size of structures is within this threshold.

**Reply by chubei-oppen** [reply]
Thanks for your answer!
